src/actions/editor.js

```diff
@@ -32,3 +32,41 @@ export const renameNode = (targetModuleIndex, targetNode, newName) => {
     }
   })
 }
+
+export const newModule = (nextModuleIndex) => {
+  return ({
+    type: 'NEW_MODULE',
+    data: nextModuleIndex
+  })
+}
+
+export const showLoadModule = () => {
```

Comment:
This and the hide could realistically be local state inside the modal since the rest of the app probably doesn't care if it's open or not.

Reply:
How do we communicate the fact that we need to open the modal when you click on the link in the nav bar?  I do it with an event that changes the application state which then flows down into a prop on the modal to set it to not be hidden.  Is there a better way to go about this?

Reply:
The trick I've used in the past is to always render the modal  but have a check inside it for `hidden` that defaults to true. If it's hidden you display the button/link to open it with an click handler to toggle state. If it's displayed you then display the modal html. 

The benefit of this approach is that you don't need things in your redux store for every single modal in your application and your modals are self contained entities.

Reply:
Oh, interesting, so the button to open the modal and the modal are in the same component?  I like that idea.  Initially it made me a little uncomfortable because I've always put the modal div in a different part of the DOM (generally right before the close of the body).  But I guess that's technically not necessary... and in researching there is also the option of using 'portals', which allow you to attach things like modals to the end of the dom tree.  It seems like the other generally accepted way of doing this (and what is recommended by Dan Abrimov) is to just do a generic 'open modal' event, which is processed by a single 'modal' reducer, which then renders a specific modal within a 'Modal Root' component that exists at the top level of your application.